### PR TITLE
Describe how to stream a GET response

### PR DESCRIPTION
**Pull request by deitch**
Describe in the README how to stream a response directly to a file or other stream handler.

As requested by #452

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -263,7 +263,28 @@ You can:
...
-See `RestClient::Request`'s documentation for more information.
+See `RestClient::Request`'s documentation for more information. 
+
+### Streaming
+
+Normally, when you use `RestClient.get` or the lower level `RestClient::Request.execute method: :get` to retrieve data, the entire response is buffered in memory and returned as the response to the call.
+
+However, if you are retrieving a large amount of data, for example a Docker image, an iso or any other large file, you may want to stream the response directly to disk rather than loading it in memory. If you have a very large file, it may become *impossible* to load it into memory.
+
+If you want to stream the data from the `GET` to a file as it comes, rather than entirely in memory, you must pass `RestClient::Request.execute` a parameter `blok_response` to which you pass a `block` that streams the request. That block, in turn, will receive the response and the ability to stream directly to file as each chunk comes across.
```
Comment (akostadinov) [inline README.md]: typo `blok_response`

Comment (deitch) [inline README.md]: Oops! :-) I will fix it.

**Comment by ab** [comment]
A very belated thanks! I admit ironically that I haven't paid much attention to streaming functionality since I haven't used it much, in part because it's so poorly documented. So thank you for the detective work and contribution.

**Comment by deitch** [comment]
You are quite welcome. I wrote a gem https://github.com/deitch/docker_registry2 that can talk directly to docker registry v2, so needed to be able to stream multi-hundred-MB images directly to storage.

**Pull request merged**